### ISSUE TITLE
AssertionError ... assert df > 0

### ISSUE THREAD

**Issue by JohnDenker** [usnistgov/SP800-90B_EntropyAssessment#12]
### Recipe:

:; time python iid_main.py -v urandom-10.bin 8
### Observed behavior:

```
** Passed IID permutation tests
Traceback (most recent call last):
  File "iid_main.py", line 66, in <module>
    if pass_chi_square_tests(dataset, verbose):
  File ".../chi_square_tests.py", line 31, in pass_chi_square_tests
    cutoff = chi_square_cutoff(df)
  File ".../chi_square_tests.py", line 309, in chi_square_cutoff
    assert df > 0
AssertionError

real    6m10.137s
user    6m5.730s
sys     0m2.420s
```

I ran the command twice and observed the assertion error both times, so it appears to be reproducible.  
See attached file for additional details.
[crash.txt](https://github.com/usnistgov/SP800-90B_EntropyAssessment/files/392528/crash.txt)

**Comment by celic** [comment]
I'd like to note that your dataset is of 10,000 bytes instead of the recommended 1 million bytes. While the time-performance of the Python is pretty rough on the larger dataset, I'm sure this assertion would pass on a fully-sized dataset. Perhaps comment out the permuation tests in `iid_main.py` to begin with the Chi Squares test and get rid of much of the wait.

**Comment by JohnDenker** [comment]
I added a pull request.
Originally the PR contained two commits.

Since then I have pushed additional commits.  Github decided to add them to the PR, which is confusing.

The third commit addresses an unrelated bug:  the test result should be independent of verbosity.

The fifth commit address the suggestion about running the expensive permutation tests **after** all the cheap and easy tests have been run.
